### ISSUE TITLE
[Petstore] Errors in petstore.json according to http://editor.swagger.io/

### ISSUE THREAD

**Issue by akhilkedia94** [swagger-api/swagger-codegen#2552]
On using the **master branch petstore.json**, swagger editor [http://editor.swagger.io/](url) gives the error - 
```

 Swagger Error
Status values that need to be considered for filter
Jump to line 132
Details
 Object
code:  "INVALID_TYPE"
message:  "Expected type array but found type string"
 path: Array [6]
description:  "Status values that need to be considered for filter"
level: 900
type:  "Swagger Error"
lineNumber: 132
```
![a](https://cloud.githubusercontent.com/assets/13836826/14426309/140894fc-0029-11e6-9190-e4437b102f22.jpg)

**Steps to reproduce**

1. copy the petstore.json from [https://raw.githubusercontent.com/swagger-api/swagger-codegen/master/modules/swagger-codegen/src/test/resources/petstore.json](url)
2. paste in swagger editor [http://editor.swagger.io/](url)

**Possible corrections**
The above error does not occur when using the json available at [http://petstore.swagger.io/v2/swagger.json](url)
The above file has an enum field in the pet/findByStatus/parameters/0/items
`              "enum":[  
                "available",
                "pending",
                "sold"
              ],`
Adding the above code to the petstore.json causes the error to disappear from swagger editor.

**Comment by wing328** [comment]
@akhilkedia94 yes, that needs to be fixed. Please use https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/test/resources/2_0/petstore.yaml for the time being.

I should be able it fix it within this week.

**Comment by ePaul** [comment]
Is this fixed by #2623?

**Comment by wing328** [comment]
@ePaul @akhilkedia94 yes. Now petstore.json and petstore.yaml are representing the same set of endpoints and models.
